app/src/ui/merge-branch/merge.tsx

```diff
@@ -143,7 +149,7 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
               Merge into <strong>{currentBranch ? currentBranch.name : ''}</strong>
             </Button>
           </ButtonGroup>
-          {mergeInfo}
+          {this.renderMergeInfo(invalidBranch)}
```

Comment:
I dont follow what you are getting at. Are you suggesting:
`{invalidBranch()}`
`{this.renderMergeInfo()}`

That would still require that renderMergeInfo() check for invalidBranch so I dont know what that really buys us